Http/Controllers/API/V1/PriceController.php

```diff
@@ -23,75 +24,77 @@ public function index(Request $request)
             'query' => 'nullable|string|max:50',
         ]);
 
-        $lists = [];
-        if ($request->has('list_id')) {
-            $lists[] = $request->get('list_id');
-        } else if ($request->has('list_ids')) {
-            $lists = $request->array('list_ids');
-        } else if ($request->has('reimbursement_type_id')) {
-            $query = TenantPriceList::forReimbursementType($request->get('reimbursement_type_id'))
-                ->used();
+        $lists = $this->buildListsArrayFromRequestParams($request);
+        $authIssueResponse = $this->authorizationCheckForLists($lists);
+        if ($authIssueResponse) {
+            return $authIssueResponse;
+        }
 
-            if ($priceList = $query->first()) {
-                $lists[] = $priceList;
-            }
+        $query  = PriceQueryHelper::buildPriceOnlyQuery($request,$lists);
+        if(empty($query)) return $this->emptyItem();
+
+        if ($request->has('query')) {
+           $query = PriceQueryHelper::applyQueryParamFromRequest($request,$query);
         }
 
-        if (count($lists)) {
-            foreach ($lists as $key => $list) {
-                $list = is_numeric($list) ? TenantPriceList::find($list) : $list;
-                $this->authorize('view', [$list, TenantPriceList::class]);
-                $lists[$key] = $list->id;
-            }
-        } else {
-            $this->authorize('view', [null, TenantPriceList::class]);
+        if ($request->has('sort_by')) {
+            $query->orderBy('prices.'.$request->get('sort_by'), $request->get('order') ?: 'asc');
         }
 
-        $query = Price::only()
-            ->forPriceLists($lists)
-            ->when($request->has('deleted'), function ($query) use ($request) {
-                return $query->where('prices.deleted', $request->boolean('deleted'));
-            });
+		return $this->paginator($query->paginate($request->get('per_page') ?: 30));
+    }
 
-        if ($request->has('code')) {
-            $query->withSocialInsuranceAgencyId($request->get('code'));
+    /**
+     * Builds an array of List IDs from request parameters.
+     *
+     * @param  \Illuminate\\Http\Request  $request
+     * @return array
+     */
+    private function buildListsArrayFromRequestParams(Request $request){
```

Comment:
In this metod we have a few mutually exclusive if statements instead of the original "chain" of if-else statements, thus the logic is preserved as before